plugin-server/src/main/pluginsServer.ts

```diff
@@ -434,12 +434,12 @@ export async function startPluginsServer(
             if (batchConsumer) {
                 stopSessionRecordingBlobConsumer = async () => {
                     // Tricky - in some cases the hub is responsible, in which case it will drain and clear. Otherwise we are responsible.
+                    await ingester.stop()
```

Comment:
might be safer over time if we have some code that enforces the order of shutdown... whether that's a test on this behaviour or some order list of shut down tasks that makes clear the order is relatively fixed.

no need to block here but "swapping the order of adjacent lines introduces a hard to see bug" is a yucky position 🤣

Reply:
Yeah trying to write a test at the moment but it is really hard to reproduce...

Reply:
Got a test in at least chhecking that no promises error. Not quite the same but close